plugin/trino-starrocks/src/main/java/io/trino/plugin/starrocks/StarRocksClient.java

```diff
@@ -0,0 +1,676 @@
+        }
+        if (type == REAL) {
+            return WriteMapping.longMapping("float", realWriteFunction());
+        }
+        if (type == DOUBLE) {
+            return WriteMapping.doubleMapping("double", doubleWriteFunction());
+        }
+        if (type instanceof DecimalType decimalType) {
+            String dataType = format("decimal(%s, %s)", decimalType.getPrecision(), decimalType.getScale());
+            if (decimalType.isShort()) {
+                return WriteMapping.longMapping(dataType, shortDecimalWriteFunction(decimalType));
+            }
+            return WriteMapping.objectMapping(dataType, longDecimalWriteFunction(decimalType));
+        }
+        if (type == DATE) {
+            return WriteMapping.longMapping("date", starRocksDateWriteFunctionUsingLocalDate());
+        }
+        if (type instanceof CharType charType) {
+            if (charType.getLength() <= 255) {
+                return WriteMapping.sliceMapping("char(" + charType.getLength() + ")", charWriteFunction());
+            }
+        }
+        if (type instanceof VarcharType varcharType) {
+            if (!varcharType.isUnbounded() && varcharType.getBoundedLength() <= 1048576) {
+                return WriteMapping.sliceMapping("varchar(" + varcharType.getBoundedLength() + ")", varcharWriteFunction());
+            }
+        }
+
+        throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type.getDisplayName());
+    }
+
+    @Override
+    public Map<String, Object> getTableProperties(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        if (!tableHandle.isNamedRelation()) {
+            return ImmutableMap.of();
+        }
+
+        try (Connection connection = connectionFactory.openConnection(session);
+                Handle handle = Jdbi.open(connection)) {
+            RemoteTableName remoteTableName = tableHandle.getRequiredNamedRelation().getRemoteTableName();
+            return handle.createQuery("SELECT TABLE_ENGINE, DISTRIBUTE_KEY, PROPERTIES FROM information_schema.tables_config " +
+                            "WHERE TABLE_SCHEMA = :schema AND TABLE_NAME = :table_name")
+                    .bind("schema", remoteTableName.getCatalogName().orElseThrow())
+                    .bind("table_name", remoteTableName.getTableName())
+                    .map((rs, ctx) -> {
+                        ImmutableMap.Builder<String, Object> tablePropertiesBuilder = ImmutableMap.builder();
+                        String engine = rs.getString("TABLE_ENGINE");
+                        checkState(!rs.wasNull(), "ENGINE is null");
+                        tablePropertiesBuilder.put("engine", StarRocksEngineType.valueOf(engine));
+
+                        String distributionDesc = rs.getString("DISTRIBUTE_KEY");
+                        if (!isNullOrEmpty(distributionDesc)) {
+                            tablePropertiesBuilder.put("distribution_desc", extractDistributionKeys(distributionDesc));
```

Comment:
why is table property called `distribution_desc` even though StarRocks itself calls it `distribute_key`?

Reply:
Yeah, seems StarRocks prefer `_desc` suffix to `_key` suffix, see [key_desc](https://docs.starrocks.io/en-us/2.5/sql-reference/sql-statements/data-definition/CREATE_TABLE#key_desc), [partition_desc](https://docs.starrocks.io/en-us/2.5/sql-reference/sql-statements/data-definition/CREATE_TABLE#partition_desc),etc.